tests/integration/test_fim/test_registry/test_restrict/test_registry_restrict.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+import os
+
+import pytest
+
+from wazuh_testing import global_parameters
+from wazuh_testing.fim import LOG_FILE_PATH, create_registry, modify_registry_value, registry_parser, KEY_WOW64_32KEY, \
+    KEY_WOW64_64KEY, callback_restricted, generate_params, callback_detect_event, delete_registry_value, delete_registry, \
+    check_time_travel
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
+from wazuh_testing.tools.monitoring import FileMonitor
+from win32con import REG_SZ
+pytestmark = [pytest.mark.win32, pytest.mark.tier(level=1)]
+
+# Variables
+key = "HKEY_LOCAL_MACHINE"
+sub_key_1 = "SOFTWARE\\testkey1"
+sub_key_2 = "SOFTWARE\\testkey"
+
+test_regs = [os.path.join(key, sub_key_1),
+             os.path.join(key, sub_key_2)]
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+reg1, reg2 = test_regs
+
+valid_subkey = "restrict_key"
+valid_value_name = "restrict_value"
+no_valid_subkey = "some_key"
+no_valid_value_name = "somme_value"
+
+# Configurations
+conf_params = {'WINDOWS_REGISTRY_1':reg1, 'WINDOWS_REGISTRY_2':reg2}
+
+configurations_path = os.path.join(test_data_path, 'wazuh_restrict_conf.yaml')
+p, m = generate_params(extra_params=conf_params, modes=['scheduled'])
+configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
+
+# Fixtures
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.mark.parametrize('key, subkey, arch, value_name, triggers_event, tags_to_apply', [
+    (key, sub_key_1, KEY_WOW64_64KEY, valid_value_name, True, {'value_restrict'}),
+    (key, sub_key_2, KEY_WOW64_32KEY, valid_value_name, True, {'value_restrict'}),
+    (key, sub_key_2, KEY_WOW64_64KEY, valid_value_name, True, {'value_restrict'}),
+    (key, sub_key_1, KEY_WOW64_64KEY, no_valid_value_name, False, {'value_restrict'}),
+    (key, sub_key_2, KEY_WOW64_32KEY, no_valid_value_name, False, {'value_restrict'}),
+    (key, sub_key_2, KEY_WOW64_64KEY, no_valid_value_name, False, {'value_restrict'})
+])
+def test_restrict_value(key, subkey, arch, value_name, triggers_event, tags_to_apply,
+                  get_configuration, configure_environment, restart_syscheckd,
+                  wait_for_fim_start):
+    """
+    Check the only files detected are those matching the restrict regex
```

Comment:
This test is about registry values, not files. Please, changes this comment.
```suggestion
    Check the only registry values detected are those matching the restrict regex.
```